src/main/java/org/dependencytrack/persistence/ProjectQueryManager.java

```diff
@@ -467,7 +472,25 @@ public Project updateProject(Project transientProject, boolean commitIndex) {
         project.setCpe(transientProject.getCpe());
         project.setPurl(transientProject.getPurl());
         project.setSwidTagId(transientProject.getSwidTagId());
-        project.setActive(transientProject.isActive());
+
+        if (project.isActive() && hasActiveChild(project) && (transientProject.isActive() == null || !transientProject.isActive()) ){
```

Comment:
```suggestion
        if (project.isActive() && !Boolean.TRUE.equals(transientProject.isActive()) && hasActiveChild(project)){
```
Auch hier wieder: hasActiveChild als letztes, da es Performance kostet und oft nicht nötig sein wird. durch !Boolean.True.equals() kannst du dir den null oder false-check sparen.